app/crash/crash.go

```diff
@@ -0,0 +1,118 @@
+// Package crash handles unexpected panics by running registered crash handlers.
+package crash
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/utils/logger"
+)
+
+// PanicInformation describes a panic.
+type PanicInformation struct {
+	ReceivedAt time.Time       `json:"received_at"`
+	Error      error           `json:"error"`
+	Trace      string          `json:"stack"`
+	Context    context.Context `json:"context"`
+}
+
+// Handler functions handle errors received by errorSignal.
+// They can be used for a number of purposes, such as logging the error, send it to external monitoring services,
+// or generate crash reports
+type Handler func(info PanicInformation)
+
+// Manager will provide Defer functions that capture panics, and handles them using registered handlers.
+type Manager struct {
+	// array of registered handlers
+	handlers []Handler
+
+	// errorSignal is a channel that lets goroutines catch their own panics and safely forward them to one central listener.
+	errorSignal chan PanicInformation
+
+	// Report collects information that should be persisted in case of a crash
+	Report *Report
+}
+
+// Default manager provides the default crash behavior, with Log and Bugsnag handlers.
+var Default *Manager
+
+// New creates a new Crash manager,
+func New() *Manager {
+	return &Manager{
+		errorSignal: make(chan PanicInformation),
+		Report: &Report{
+			Metadata: make(map[string]interface{}),
+		},
+	}
+}
+
+func init() {
+	// Setup integration with https://www.bugsnag.com error monitoring service
+	SetupBugsnag()
+
+	Default = New()
+	Default.RegisterHandler(Default.Report.ReportHandler)
```

Comment:
Should we have a config.toml variable to turn off ReportHandler. Can be useful in dev environments or for those who do not want to generate crash reports